### DISCUSSION TITLE
Criteria for stable release

### DISCUSSION THREAD

**Discussion by Kevsy** [camaraproject/EdgeCloud#170] [General]
@sergiofranciscoortiz  here are the items I believe we need to check before each release:

1. Confirm conformance with API Design Guidelines: run the [CAMARA linter locally](https://github.com/ravindrapalaskar17/Commonalities/blob/API-linting-Implementation-Guideline/documentation/API-linting-Implementation-Guideline.md#api-linting-configuration-steps-for-local) and fix any errors reported. 
2. Add the [CAMARA Consent code and documentation](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#camara-api-specification---authorization-and-authentication-common-guidelines)
3. Check for any common datatypes to reuse from [common.json](https://github.com/ravindrapalaskar17/Commonalities/blob/API-linting-Implementation-Guideline/artifacts/CAMARA_common.json)
4. A Gherkin `.feature` file in Test definitions
5. Include a User Story based on the Commonalities [User Story template](https://github.com/camaraproject/Commonalities/blob/main/documentation/Userstory-template.md)

**Comment by sergiofranciscoortiz** [comment]
Thanks for the details @Kevsy.

We were trying to follow steps indicated on first link to run linter locally, there is a link to [.spectral.yml](https://github.com/camaraproject/Commonalities/blob/main/artifacts/linting_rules/.spectral.yml) for which we are getting a 404, apparently the definition of rulesets are in progress in https://github.com/camaraproject/Commonalities/pull/74 

From there we could obtain a [draft spectral ](https://github.com/camaraproject/Commonalities/blob/aaeb455e3400347de1ae6558337ef739cb943837/artifacts/linting_rules/.spectral.yml )and lint function files. After a first run we have detected a few things to change in EdgeCloud_LcM.yaml to comply with kebab-case in URLs ( we will be opening a pull request to sort that out). 
We noticed, than in general the reponse consider errors in all paths even if they comply with the kebab-case pattern, we would say that it is considering the leading slash of the path when it shouldn´t but we don´t know how to avoid this in spectral.yml definition

```
  71:16    error  camara-parameter-casing-convention  /appInstance should be kebab-case: must be kebab case                                                     paths./appInstance          
  212:8    error  camara-parameter-casing-convention  /app should be kebab-case: must be kebab case                                                             paths./app
 331:21    error  camara-parameter-casing-convention  /edge-cloud-nodes should be kebab-case: must be kebab case                                                paths./edge-cloud-nodes
```

In above example /app and /edge-cloud-nodes follow kebab case and are shown as casing convention errors (when they shouldn´t).

**Reply by Kevsy** [reply]
Good point- can you report that back to Commonalities? I did not get those errors for [Connectivity Insights](https://github.com/camaraproject/ConnectivityInsights/discussions/23) but maybe that was with an earlier version of the spectral draft.

**Comment by Kevsy** [comment]
Added the User Story template (5)
